modin/utils.py

```diff
@@ -58,15 +64,143 @@ def _get_indent(doc: str) -> int:
     int
         Minimal indent (excluding empty lines).
     """
+    indents = _get_indents(doc)
+    return min(indents) if indents else 0
+
+
+def _get_indents(source: Union[list, str]) -> list:
+    """
+    Compute indentation for each line of the source string.
+
+    Parameters
+    ----------
+    source : str or list of str
+        String to compute indents for. Passed list considered
+        as a list of lines of the source string.
+
+    Returns
+    -------
+    list of ints
+        List containing computed indents for each line.
+    """
     indents = []
-    for line in doc.splitlines():
+
+    if not isinstance(source, list):
+        source = source.splitlines()
+
+    for line in source:
         if not line.strip():
             continue
         for pos, ch in enumerate(line):
             if ch != " ":
                 break
         indents.append(pos)
-    return min(indents) if indents else 0
+    return indents
+
+
+def format_string(template: str, **kwargs) -> str:
+    """
+    Insert passed values at the corresponding placeholders of the specified template.
+
+    In contrast with the regular ``str.format()`` this function computes proper
+    indents for the placeholder values.
+
+    Parameters
+    ----------
+    template : str
+        Template to substitute values in.
+    **kwargs : dict
+        Dictionary that maps placeholder names with values.
+
+    Returns
+    -------
+    str
+        Formated string.
+    """
+    # We want to change indentation only for those values, whose placeholder located
+    # at the beginning of the line, in that case, the placeholder sets an indentation
+    # that the filling value has to obey.
+    template_lines = template.splitlines()
+    # RegExp determining placeholders located at the beginning of the line.
+    regex = r"^[ ^{]*{(\w+)}"
```

Comment:
```suggestion
    regex = r"^ *{(\w+)}"
```
your form of `[ ^{]` is not matching your intent (I think you wanted to match "space but not curly brace", but `^` means "not" in regex _only_ if it's the first symbol after square bracket, otherwise it's just a normal symbol).